File: android/lib/interfaces/event-description.dart
import 'package:polydiff/pages/replay-page.dart';

import 'observer-event-description.dart';

abstract class EventDescription {
  final int time;
  final double? x=null;
  final double? y=null;
  final List<int>? userCounter=null;
  final Rectangle? rectangle=null;
  final String? color=null;

  EventDescription.fromJson(Map<String, dynamic> json)
      : time = json['time'];



  const EventDescription(this.time);
  play(ReplayPage replayPage);
}

File: client/src/app/classes/event-description.ts
import { ReplayPageComponent } from "@app/pages/replay-page/replay-page.component";

export abstract class EventDescription {
    time: number;

    constructor(time: number) {
        this.time = time;
    }

    play(replayPage: ReplayPageComponent): void{
        console.log("Playing event at time: " + this.time);
    };
}	
